### PR TITLE
Add live tests exercising lag option

### PR DESCRIPTION
**Pull request by bkutil**

### CODE REVIEW

**Comment by bkutil** [comment]
cc: @dmehra

File: test/influx-integration.spec.js [hunk 1]
```diff
@@ -435,7 +435,7 @@ describe('@integration influxdb tests', () => {
 
             it('successfully retrieves live points', () => {
                 return check_juttle({
-                    program: 'read influx -db "test" -from :now: -to :end: -every :0.1s: name = "live" | view logger',
+                    program: 'read influx -db "test" -to :end: -every :0.1s: name = "live" | view logger',
```
Comment (dmehra) [inline test/influx-integration.spec.js]: i don't follow, you removed the explicit `-from :now:` but that's the default, it should be equivalent?

File: test/influx-integration.spec.js [hunk 2]
```diff
@@ -444,6 +444,34 @@ describe('@integration influxdb tests', () => {
             });
         });
 
+        describe('lag', () => {
+            it('successfully retrieves lagging points', () => {
+                // Start inserting points every 0.1s with 0.5s timestamp lag
```
Comment (dmehra) [inline test/influx-integration.spec.js]: is it reliable to use 0.5s here and the same value for `-lag` or should one be larger than the other?

File: test/influx-integration.spec.js (hunk 1 above)
Comment (bkutil) [inline test/influx-integration.spec.js]: This unrelated change crept in while I was playing with `lag` test. Removed.

**Comment by dmehra** [comment]
How about a second test where the read uses `-lag 0.5s` but the points are arriving without delay (current timestamps), just as a sanity check? After waiting 0.5s we should have those points.

File: test/influx-integration.spec.js (hunk 2 above)
Comment (bkutil) [inline test/influx-integration.spec.js]: A certain margin wouldn't hurt, but I think that same numbers make the test easier to understand. If we see 'random' failures, then lets increase this.

**Comment by bkutil** [comment]
Would that exercise anything extra compared to this test?

File: test/influx-integration.spec.js (hunk 2 above)
Comment (bkutil) [inline test/influx-integration.spec.js]: lol :) OK, lets increase it :)

**Comment by rlgomes** [comment]
Why not do the same tests with a few seconds instead of fractions of seconds like I did here:

https://github.com/juttle/juttle-graphite-adapter/blob/master/test/graphite.spec.js#L93

And the only reason to do multi seconds is that you can write points into the near future and see them show up as expected and it also verifies existing lag or waht the real `lag` value should be for something like influx. I know for graphite that the default lag we have right now "barely" works and you'd have to run with a :5s: to really get live to work for graphite.

**Comment by bkutil** [comment]
> Why not do the same tests with a few seconds instead of fractions of seconds like I did here:

I initially tested with whole seconds, then trimmed down the time so that the test doesn't take too long.

> and it also verifies existing lag or waht the real lag value should be for something like influx.

Could you please explain? You mean adjusting the lag to DB latency?

**Comment by rlgomes** [comment]
@bkutil yes the db latency as elastic and graphite have a few seconds (depending on load and setup) before the thing you just stored is readable. I'm not sure of what that lag is like for influx and could be that the current :1s: lag works fine.

**Comment by bkutil** [comment]
I.e. the value for lag option should serve as a documentation for the 'time to write' for a given database.

It seems that the subsecond lag is fine for influx.

**Comment by bkutil** [comment]
@rlgomes @dmehra bump. If you're OK with the state of this PR, let's merge it.

**Comment by rlgomes** [comment]
+1 didn't mean to delay the PR but I do think we need to document or better understand what it means to test live for each of the different adapters.
